iroha_p2p/src/peer.rs

```diff
@@ -454,130 +731,63 @@ where
 #[async_trait::async_trait]
 impl<T, K, E> Handler<Post<T>> for Peer<T, K, E>
 where
-    T: Encode + Decode + Send + Sync + Clone + 'static,
+    T: Debug + Encode + Decode + Send + Sync + Clone + 'static,
     K: KeyExchangeScheme + Send + 'static,
     E: Encryptor + Send + 'static,
 {
     type Result = ();
 
-    async fn handle(&mut self, msg: Post<T>) {
-        if self.write.is_none() {
-            warn!("Cannot send message to peer, as we are not connected!");
-            return;
-        }
-
-        let data = match &self.cipher {
-            None => msg.data.encode(),
-            Some(cipher) => match cipher.encrypt_easy(DEFAULT_AAD.as_ref(), &msg.data.encode()) {
-                Ok(data) => data,
-                Err(e) => {
-                    warn!(%e, "Error encrypting message!");
-                    self.state = State::Error;
-                    return;
-                }
-            },
-        };
+    async fn handle(&mut self, Post { data, .. }: Post<T>) {
         debug!("Sending message: {:?}", &data);
+        let data = match encrypt_easy(&self.cipher, &data.encode()) {
+            Ok(data) => data,
+            Err(err) => {
+                debug!(%err, "Encryption error!");
+                return;
+            }
+        };
         #[allow(clippy::unwrap_used)]
-        let mut write_half = self.write.as_mut().unwrap();
-        if let Err(e) = send_message(&mut write_half, data.as_slice()).await {
+        if let Err(e) = self.send_message(data.as_slice()).await {
             warn!(%e, "Error sending message to peer!");
-            self.state = State::Error;
```

Comment:
We can assume that this connection is broken/disconnected and we need to stop this peer.